coordinator/src/main/java/com/tencent/rss/coordinator/SimpleClusterManager.java

```diff
@@ -72,14 +72,17 @@ public SimpleClusterManager(CoordinatorConf conf) {
     }
   }
 
-  private void nodesCheck() {
+  void nodesCheck() {
     try {
       long timestamp = System.currentTimeMillis();
       Set<String> deleteIds = Sets.newHashSet();
       for (ServerNode sn : servers.values()) {
         if (timestamp - sn.getTimestamp() > heartbeatTimeout) {
           LOG.warn("Heartbeat timeout detect, " + sn + " will be removed from node list.");
           deleteIds.add(sn.getId());
+          for (Set<ServerNode> nodesWithTag : tagToNodes.values()) {
+            nodesWithTag.remove(sn);
+          }
         }
       }
       for (String serverId : deleteIds) {
```

Comment:
for (String serverId : deleteIds) {
      ServerNode node = servers.remove(serverId);
      if (node != null) {
           for (Set<ServerNode> nodesWithTag : tagToNodes.values()) {
                 nodesWithTag.remove(sn);
          }
      }
}
In the such way, we can put two places deletion logic together.

Reply:
Yep. It looks better.